src/ConnectedMode/UI/AutomaticBindingRequest.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI;
+
+public abstract record AutomaticBindingRequest
+{
+    internal abstract string TypeName { get; }
+
+    private AutomaticBindingRequest() { }
+
+    public record Shared : AutomaticBindingRequest
+    {
+        public static readonly AutomaticBindingRequest Current = new Shared();
+        private Shared() { }
+
+        internal override string TypeName => ConnectedMode.Resources.AutomaticBindingType_Shared;
+    }
+
+    public record Assisted(
```

Comment:
I think here it should be "Automatic"

```suggestion
    public record AutomaticBindingRequest(
```